src/scripts/main.js

```diff
@@ -1,3 +1,259 @@
+  const inputPosition = employee.querySelector('[data-qa="position"]');
+  const inputOffice = employee.querySelector('[data-qa="office"]');
+  const inputAge = employee.querySelector('[data-qa="age"]');
+  const inputSalary = employee.querySelector('[data-qa="salary"]');
+
+  const valuesOfInputs = [
+    inputName,
+    inputPosition,
+    inputOffice,
+    inputAge,
+    inputSalary,
+  ];
+
+  if (
+    inputName.value.trim() !== '' &&
+    inputName.value.length >= 4 &&
+    inputAge.value.trim() !== '' &&
+    parseInt(inputAge.value) >= 18 &&
+    parseInt(inputAge.value) <= 90 &&
+    inputPosition.value !== '' &&
+    inputSalary.value !== ''
+  ) {
+    const row = document.createElement('tr');
+
+    valuesOfInputs.forEach((input) => {
+      const cell = document.createElement('td');
+
+      if (input.getAttribute('data-qa') === 'salary') {
+        cell.textContent = formatCurrency(input.value);
+      } else {
+        cell.textContent = input.value;
+      }
+
+      row.appendChild(cell);
+    });
+
+    tbody.appendChild(row);
+
+    notificationMessage('Adding successfully', 'success');
+
+    employee.reset();
+  } else {
+    if (inputName.value.length < 4) {
+      notificationMessage('Name length is less than 4 characters', 'error');
+    }
+
+    if (parseInt(inputAge.value) < 18) {
+      notificationMessage('Your age is too young :(', 'error');
+    }
+
+    if (parseInt(inputAge.value) > 90) {
+      notificationMessage('Your age is too old :(', 'error');
+    }
+
+    if (inputPosition.value.trim() === '') {
+      notificationMessage('Indicate position :(', 'error');
+    }
+  }
+
+  e.preventDefault();
+});
+
+// #endregion
+
+// #region sorting rows
+function sortedBodyRows(rows, index, nameOfCell) {
+  if (nameOfCell === 'Age' || nameOfCell === 'Salary') {
+    rows.sort(
+      (a, b) =>
+        stringToNumber(a.children[index].textContent) -
+        stringToNumber(b.children[index].textContent),
+    );
+  } else {
+    rows.sort((a, b) => {
+      return a.children[index].textContent.localeCompare(
+        b.children[index].textContent,
+      );
+    });
+  }
+
+  return rows;
+}
+
+// #endregion
+
+// #region formatting a string to a number
+function stringToNumber(str) {
+  if (str.includes(',')) {
+    return parseInt(
+      str
+        .split('')
+        .filter((el) => Number.isInteger(parseInt(el)))
+        .join(''),
+    );
+  } else {
+    return parseInt(str);
+  }
+}
+// #endregion
+
+// #region activity class switch
+[...tbody.querySelectorAll('tr')].forEach((row, index) => {
+  row.addEventListener('click', () => {
+    if (indexOfRow !== index) {
+      [...tbody.querySelectorAll('tr')].forEach((element) => {
+        element.classList.remove('active');
+      });
+    }
+    indexOfRow = index;
+
+    row.classList.add('active');
+  });
+});
+// #endregion
+
+// #region sorting in two directions
+cellsOfHead.forEach((cell, index) => {
```

Comment:
use [event delegation](https://www.freecodecamp.org/news/event-delegation-javascript/) instead of adding event to each header separately using loop